src/audio/base_fw.c

```diff
@@ -455,16 +461,86 @@ static int basefw_set_fw_config(bool first_block,
 	return 0;
 }
 
+int schedulers_info_get(uint32_t *data_off_size,
+			char *data,
+			uint32_t core_id)
+{
+	/* TODO
+	 * Core id parameter is not yet used. For now we only get scheduler info from current core
+	 * Other cores info can be added by implementing idc request for this data.
+	 * Do this if Schedulers info get ipc has uses for accurate info per core
+	 */
+
+	struct scheduler_props *scheduler_props;
+	/* the internal structs have irregular sizes so we cannot use indexing, and have to just
+	 *  to reassign pointers for each element
+	 */
+	struct schedulers_info *schedulers_info = (struct schedulers_info *)data;
+
+	schedulers_info->scheduler_count = 0;
+
+	/* smallest response possible is just zero schedulers count
+	 * here we replace max_len from data_off_size to serve as output size
+	 */
+	*data_off_size = sizeof(struct schedulers_info);
+
+	/* ===================== LL_TIMER SCHEDULER INFO ============================ */
+	schedulers_info->scheduler_count++;
+	scheduler_props = (struct scheduler_props *)(data + *data_off_size);
+	scheduler_get_task_info_ll(scheduler_props, data_off_size);
+
+	/* ===================== LL_DMA SCHEDULER INFO ============================ */
+	/* Could not make this one work */
+	/* schedulers_info->scheduler_count++;
+	 * scheduler_props = (struct scheduler_props *)(data + *data_off_size);
+	 * scheduler_get_task_info_ll_dma(scheduler_props, data_off_size);
+	 */
```

Comment:
Did not manage to get any info from LL_DMA scheduler. I was not sure what type of struct scheduler_get_data(SOF_SCHEDULE_LL_DMA) would return. Are there any tasks there anyway?

Reply:
I would call the Zephyr (low level data) & SOF (high level data) APIs here to get info on LL and DP schedulers as 1st step. I dont think we have any users of the DMA driven scheduling on Zephyr.

Reply:
Yeah i have some info on those two. Removing LL DMA info from here for now.